esp-hal/src/dma/mod.rs

```diff
@@ -1936,6 +1936,16 @@ impl DmaTxBuf {
         Ok(buf)
     }
 
+    /// Create an empty DmaTxBuf.
+    ///
+    /// This has no ability to transmit data.
+    pub fn empty() -> Self {
+        Self {
+            descriptors: &mut [],
```

Comment:
I forgot to document this invariant but even an empty buffer requires at least one descriptor.

Side note: Are you confident that `&mut []` will end up in DRAM?

Reply:
Yeah I just found this one :D. IIRC anything `&mut` will end up in ram

Reply:
> I forgot to document this invariant but even an empty buffer requires at least one descriptor.

Is this a limitation of the hardware DMA impl, or can we work around this with an early exit somewhere? It turns out its quite annoying to have one item in there.

Reply:
This is indeed a hardware limitation, the DMA channels require a descriptor to start with, even if it's empty.

Early exit is possible but imo a bit awkward. Might be better to offer SPI APIs that don't use the data phase.
There's also the option of having a single `static` descriptor that can be used for all empty buffers.
So long as the descriptor has it's owner bit and size set to zero, we don't have to worry about mutation since the descriptor values won't change. This would have to be a separate buf type though.

Reply:
I've removed the `empty` method for now, we can revisit later.